### PR TITLE
HPCC-28353 Std.File.GetNoCommonDefault() doesn't pick up the value of noCommon from environment.xml.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Add missing xpath

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Tested manually

### CODE REVIEW

File: plugins/fileservices/fileservices.cpp
Comment (ghalliday) [inline plugins/fileservices/fileservices.cpp:3162]: @AttilaVamos did you test this?
@jakesmith Does this work?  I thought it needed to be "Software/Globals/@noCommon"

**Review by ghalliday — CHANGES_REQUESTED** [review]

File: plugins/fileservices/fileservices.cpp
Comment (jakesmith) [reply plugins/fileservices/fileservices.cpp:3162]: should be "Software/Globals/@noCommon"

@AttilaVamos - it does suggest the previous PR was not adequately tested? (i.e. need to make sure an environment setting change does actually get picked up and used)

Comment (jakesmith) [reply plugins/fileservices/fileservices.cpp:3162]: Also, I suspect this (and similar easy to spot issues in previous PR) would be spotted by you, if you scanned the changes after you issued the PR, i.e the PR issuer should always review their own PR 1st to spot issues.
As per the devdoc/CodeReview.md document:
```
* Check for silly mistakes\
  Scan the code after creating the PR to check there are no tracing/other debugging artifacts or typos.  It doesn't take long, and often catches trivial issues.  It can avoid the need for a cycle of code-review/fixes.
```

**Review by jakesmith — CHANGES_REQUESTED** [review]
@AttilaVamos - see comments

File: plugins/fileservices/fileservices.cpp
Comment (AttilaVamos) [reply plugins/fileservices/fileservices.cpp:3162]: Yes, I tested it manually with change the value of the  "noCommon" attribute in envirionment.xml, then restart HPCC and the changed value appeared in DFU Server log.

**Comment by ghalliday** [comment]
@jakesmith see code in getTailIdLength().  I think it may work even though I didn't expect it to.

File: plugins/fileservices/fileservices.cpp
Comment (jakesmith) [reply plugins/fileservices/fileservices.cpp:3162]: ok, yes I meant to say simplify by removing unecessary Owned object, didn't mean to change path used.

**Comment by AttilaVamos** [comment]
@ghalliday @jakesmith  I opened a 8.10.0 targeted PR for this fix: https://github.com/hpcc-systems/HPCC-Platform/pull/16603

**Comment by jakesmith** [comment]
> @jakesmith see code in getTailIdLength(). I think it may work even though I didn't expect it to.

hmmm, yes, I think it may, but should not.
That should be fixed (so it doesn't work). I'll open a separate JIRA (when track is working again)

**Comment by AttilaVamos** [comment]
@jakesmith so I should change the 'noCommon' to 'nocommon' and wait until your fix will be merged?

**Comment by jakesmith** [comment]
> @ghalliday @jakesmith I opened a 8.10.0 targeted PR for this fix: #16603

@AttilaVamos  better to retarget this one PR than open a fresh one.

**Comment by jakesmith** [comment]
> @jakesmith so I should change the 'noCommon' to 'nocommon' and wait until your fix will be merged?

my fix is unrelated. It is to stop blah@something being treated as blah/@something.

Afaik "noCommon" (cased) is correct, that is what dfuserver is expecting, the only place afaik that it is forced to lower case and picked up separately is from the dfu command line (where it comes in a uncased IProperties tree), and uses that option to set the spray request field.

**Comment by AttilaVamos** [comment]
How can I change the base back from master to 8.10.0 without messing-up?

**Comment by jakesmith** [comment]
> How can I change the base back from master to 8.10.0 without messing-up?

You have to rebase your branch and re-push.

e.g. 
git rebase --onto upstream/candidate-8.10.x <SHA>

where SHA is e.g. HEAD^^ if your branch has 2 commits ..

**Review by jakesmith — APPROVED** [review]
@AttilaVamos - looks correct, please squash

**Comment by AttilaVamos** [comment]
@jakesmith @ghalliday squashed.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**
